ff/src/fields/macros.rs

```diff
@@ -618,43 +618,39 @@ macro_rules! impl_Fp {
             type Output = Self;
 
             #[inline]
-            fn add(self, other: &Self) -> Self {
-                let mut result = self.clone();
-                result.add_assign(other);
-                result
+            fn add(mut self, other: &Self) -> Self {
```

Comment:
I'm confused as to why this is valid, I thought `Add` isn't supposed to mutate the base

Reply:
Here `self` is passed by value (so either the input gets moved, or copied):
```rust
let a = 2;
let c = a.add(&b); // here `a` is copied
print!(a); // will print 2
```

Reply:
Oh cool, that makes sense!